configbp/config.go

```diff
@@ -0,0 +1,94 @@
+// Package configbp parses configurations per the baseplate spec.
+package configbp
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+
+	"github.com/reddit/baseplate.go/log"
+)
+
+// BaseplateConfigPath points to the default config file per baseplate.spec.
+var BaseplateConfigPath = os.Getenv("BASEPLATE_CONFIG_PATH")
+
+type envsubstReader struct {
+	buffer bytes.Buffer
+	lines  bufio.Scanner
+}
+
+func (r *envsubstReader) Read(buf []byte) (int, error) {
+	// Keep flushing pending data if we have it
+	if r.buffer.Len() > 0 {
+		return r.buffer.Read(buf)
+	}
+
+	// Fill the buffer with some data
+	if r.lines.Scan() {
+		r.buffer.WriteString(os.ExpandEnv(r.lines.Text()))
+		r.buffer.WriteString("\n")
+	} else {
+		return 0, io.EOF
+	}
+
+	// Return some data to satisfy the reader
+	return r.buffer.Read(buf)
+}
+
+// ParseStrictFile parses configuration from the file at the given path.
+//
+// Environment variables (e.g. $FOO and ${FOO}) are substituted from the environment before parsing.
+// The configuration is parsed into each of the targets, which will typically be pointers to structs.
+func ParseStrictFile(path string, ptr interface{}) error {
+	f, err := os.Open(path)
```

Comment:
I think we still want to use `limitopen.Open` instead of `os.Open` here. Although you implemented the line reader (nice!), if the file is mapped to `/dev/zero` we could still have a single line with unlimited length that blows out memory.

Between checking for size and checking for extension (we can do both, sure, but that's overkill imo) I still slightly prefer to check size. 

We cannot use `limitopen.OpenWithLimit` directly here because the gauge and the sentry are not initialized yet, but we do can use `limitopen.Open` here.

Reply:
bufio.Scanner has an internal maximum token size of 64KiB

https://cs.opensource.google/go/go/+/refs/tags/go1.17:src/bufio/scan.go;l=81;bpv=1;bpt=1

Reply:
oh, but we will still carry it over.  I see.